### PR TITLE
PORTALS-2283

### PR DESCRIPTION
**Pull request by yjyoo773**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,6 +33,7 @@
     "dagre": "^0.8.5",
     "downshift": "^6.1.2",
     "gh-pages": "^3.2.3",
+    "html-react-parser": "^3.0.4",
```
Comment (nickgros) [inline package.json:36]: Is this necessary instead of using `dangerouslySetInnerHTML`?

File: src/lib/containers/CertificationQuiz.tsx [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+const CertificationQuiz: React.FunctionComponent = () => {
+  const { accessToken } = useSynapseContext()
+  const [quiz, setQuiz] = useState<Quiz | undefined>()
+  const [questionResponse, setQuestionResponse] = useState<QuestionResponse[]>(
+    [],
+  )
+  const [passingRecord, setPassingRecord] = useState<PassingRecord>()
+
+  const gettingStartedUrl =
+    'https://help.synapse.org/docs/Getting-Started.2055471150.html'
```
Comment (nickgros) [inline src/lib/containers/CertificationQuiz.tsx:29]:
```suggestion
  const GETTING_STARTED_URL =
    'https://help.synapse.org/docs/Getting-Started.2055471150.html'
```

File: src/lib/containers/CertificationQuiz.tsx [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    } catch (err: any) {
+      console.log(err)
+    }
+  }
+
+  useEffect(() => {
+    if (accessToken) {
+      getQuiz()
+    }
+  }, [accessToken])
```
Comment (nickgros) [inline src/lib/containers/CertificationQuiz.tsx:43]: IMO all of the state/effects are cleaner if we used `react-query`, but I recognize that you would have to manually configure the options and manipulate the cache afterwards so the quiz doesn't get overwritten. So I'll leave it up to you whether to keep what you have or make the change.

```tsx
const { data: quiz } = useGetCertifyQuiz({
  enabled: !!accessToken,
  staleTime: Infinity,
  cacheTime: Infinity
})
```

File: package.json (hunk 1 above)
Comment (yjyoo773) [reply package.json:36]: No it is not. Would you recommend using `dangerouslySetInnerHTML` over the parser?

File: src/lib/containers/CertificationQuiz.tsx [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+
+  useEffect(() => {
+    if (!accessToken) {
+      new Error('Please sign in to take certification quiz')
+    }
+  }, [accessToken])
```
Comment (nickgros) [inline src/lib/containers/CertificationQuiz.tsx:49]: The error isn't thrown. Also, the call to the certifiedUserTest endpoint throws a 403 if the user isn't signed in. Could we just do this? -

1. make the API call regardless of whether or not the user has a token
2. Then pass the error to the error boundary either with ReactQuery's `useErrorBoundary: true` or `react-error-boundary`'s useErrorHandler().handleError.
3. The error boundary will render the appropriate alert

File: src/lib/containers/CertificationQuiz.tsx [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+        answerIndex: [answer],
+        concreteType: MULTICHOICE_RESPONSE_CONCRETE_TYPE_VALUE,
+      },
+    ]
+    setQuestionResponse(newState)
+  }
+
+  const handleRetakeQuiz = () => {
+    window.location.reload()
+  }
```
Comment (nickgros) [inline src/lib/containers/CertificationQuiz.tsx:65]: Not great UX, could we just clear state (or if using react-query, clear cache) and refetch a new quiz?

File: src/__tests__/lib/containers/CertificationQuiz.test.tsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+const getQuizHandler = rest.get(
+  `${getEndpoint(
+    BackendDestinationEnum.REPO_ENDPOINT,
+  )}/repo/v1/certifiedUserTest`,
+  async (req, res, ctx) => {
+    const response: Quiz = {
+      id: 123,
+      header: 'Mock Certification Quiz',
+      questions: [],
```
Comment (nickgros) [inline src/__tests__/lib/containers/CertificationQuiz.test.tsx:31]: Would like to see 1-2 mock questions in here. Would want to see these additional scenarios: 
- Choose an answer, verify that the selected answer was submitted
- Simulate case where submit clicked and not all questions have been answered
- Simulate case where all answers are correct
- Simulate case where an answer was wrong

**Review by nickgros — CHANGES_REQUESTED** [review]
See comments

File: package.json (hunk 1 above)
Comment (nickgros) [reply package.json:36]: If it works without issues, yes. I see no reason to 
- include a new dependency
- parse the HTML that we already expect to be well-formed

File: src/lib/containers/CertificationQuiz.tsx [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+          </Typography>
+          {passingRecord.passed ? (
+            displayToast(
+              `You passed the Synapse Certification Quiz on ${passingRecord.passedOn}`,
+              'success',
+            )
+          ) : (
+            <Typography variant="body1">
+              Please review the items shown in red below, and{' '}
+              <a style={{ color: '#407ba0' }} onClick={handleRetakeQuiz}>
```
Comment (nickgros) [inline src/lib/containers/CertificationQuiz.tsx:116]: You can get the standardized link styling if you have an href. Then preventDefault so it doesn't do anything
```suggestion
              <a href="#" onClick={e => { 
                e.preventDefault()
                handleRetakeQuiz()
              }}>
```

File: src/__tests__/lib/containers/CertificationQuiz.test.tsx [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+
+    await waitFor(() =>
+      expect(mockToastFn).toBeCalledWith(
+        'Please answer all of the questions and try again.',
+        'warning',
+      ),
+    )
+  })
+
+  it('Submit quiz that did not pass', async () => {
```
Comment (nickgros) [inline src/__tests__/lib/containers/CertificationQuiz.test.tsx:93]: Can you add onto this test or write a new test to verify the `handleRetakeQuiz` works?

**Review by nickgros — CHANGES_REQUESTED** [review]
Looking good! Just a few more changes

**Review by nickgros — APPROVED** [review]
Great work @yjyoo773 !

**Pull request merged**